javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -462,8 +462,63 @@ object Character {
   //def getDirectionality(c: scala.Char): scala.Byte
 
   /* Conversions */
-  def toUpperCase(c: scala.Char): scala.Char = c.toString.toUpperCase().charAt(0)
-  def toLowerCase(c: scala.Char): scala.Char = c.toString.toLowerCase().charAt(0)
+  def toUpperCase(ch: Char): Char = toUpperCase(ch.toInt).toChar
+
+  def toUpperCase(codePoint: scala.Int): scala.Int = {
+    codePoint match {
+      case 0x1fb3 | 0x1fc3 | 0x1ff3 =>
+        (codePoint + 0x0009)
+      case _ if codePoint >= 0x1f80 && codePoint <= 0x1faf =>
+        (codePoint | 0x0008)
+      case _ =>
+        val upperChars = _String.fromCodePoint(codePoint).toUpperCase()
+        upperChars.length match {
+          case 1 =>
+            upperChars.charAt(0).toInt
+          case 2 =>
+            val high = upperChars.charAt(0)
+            val low = upperChars.charAt(1)
+            if (isSurrogatePair(high, low))
+              toCodePoint(high, low)
+            else
+              codePoint
+          case _ =>
+            codePoint
+        }
+    }
+  }
+
+/*
+for (ch <- 0.toChar until Character.MAX_VALUE) {
+  val strLowerCase: String = ch.toString.toLowerCase()
+  val chrLowerCase: Char = Character.toLowerCase(ch)
+
+  if (strLowerCase.charAt(0) != chrLowerCase)
+    println(s"    case '$ch' => '$chrLowerCase'")
+}
+// No Results
+*/
```

Comment:
This comment is not necessary anymore. We've established that it is in fact true for the 3 methods: `toUpperCase`, `toLowerCase` and `toTitleCase`. Writing this comment here but not for the other two is therefore misleading. We could also write that kind of comment on the other two instead, but that would be completely redundant with the exhaustive tests, so that's not useful IMO.